Mathlib/Data/Polynomial/Basic.lean

```diff
@@ -291,18 +291,27 @@ instance natCast : NatCast R[X] :=
 #align polynomial.has_nat_cast Polynomial.natCast
 
 instance semiring : Semiring R[X] :=
```

Comment:
Would building up through the hierarchy to `Semiring R[X]` yield even better performance? Or would it be middling?

Reply:
Same with other ones below.

Reply:
Possibly. I don't see why it would but it is certainly possible that it could because I don't fully understand this stuff

Reply:
How about merging this and I experiment with further improvements in another PR?

Reply:
Sounds good to me.